sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/Earcut.cs

```diff
@@ -645,13 +645,22 @@ private static Node insertNode(int i, float x, float y, Node last)
 
 		public static Data Flatten(List<List<float[]>> data)
 		{
-			var dim = data[0][0].Length;
-			var result = new Data() { Dim = dim };
+			var dataCount = data.Count;
+			var dim = 0;
+			for (int i = 0; i < dataCount; i++)
+			{
+				dim += data[i].Count;
+			}
+
+			var result = new Data() { Dim = 2 };
```

Comment:
Why `Dim = 2` ? If we are setting `Dim` to a hardcoded value, what is the use of the for-loop above to compute `dim` ?